### PR TITLE
Add telemetry for immersive mode duration time.

### PR DESCRIPTION
**Pull request by daoshengmu**
We would like to add a telemetry to help us figure out the avg. time spent when users in the immersive mode.

@pocmo Please help review the telemetry part. I am going to ask for the data review shortly.

### CODE REVIEW

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 1]
```diff
@@ -471,6 +477,14 @@ public void run() {
         }, 20);
     }
 
+    void uploadImmersiveElapsedTime() {
```
Comment (bluemarvin) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: This function is being called in the render thread. Is TelemetryWrapper.addImmersiveToHistogram() thread safe?

Comment (daoshengmu) [reply app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: yep. It looks risky to put it in the render thread. I have moved it to the main thread.

File: app/src/common/shared/org/mozilla/vrbrowser/telemetry/TelemetryWrapper.java [hunk 2]
```diff
@@ -184,19 +200,30 @@ public static void addLoadToHistogram(String uri, Long newLoadTime) {
 
         domainMap.add(UrlUtils.stripCommonSubdomains(uriLink.getHost()));
         numUri++;
-        int histogramLoadIndex = toIntExact(newLoadTime / BUCKET_SIZE_MS);
+        int histogramLoadIndex = toIntExact(newLoadTime / LOADING_BUCKET_SIZE_MS);
 
         if (histogramLoadIndex > (HISTOGRAM_SIZE - 2)) {
             histogramLoadIndex = HISTOGRAM_SIZE - 1;
+            Log.e(LOGTAG, "the loading histogram size is overflow.");
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/telemetry/TelemetryWrapper.java]: nit: `is overflown`

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 3]
```diff
@@ -180,6 +183,7 @@ protected void onStop() {
     @Override
     protected void onPause() {
         if (mIsPresentingImmersive) {
+            uploadImmersiveElapsedTime();
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: I'd move this function to `pauseGeckoViewCompositor`. Immersive mode can be stopped in other situations (e.g. via link traversal) apart from `onBack` & `onPause`

Comment (daoshengmu) [reply app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: @MortimerGoro IIUC, `pauseGeckoViewCompositor ` is the function to begin the immersive mode, so I put my start timer there. `onBack ` & `onPause ` are existing the immersive mode, so I measure the time spent and `uploadImmersiveElapsedTime `.

File: app/src/common/shared/org/mozilla/vrbrowser/telemetry/TelemetryWrapper.java (hunk 2 above)
Comment (daoshengmu) [reply app/src/common/shared/org/mozilla/vrbrowser/telemetry/TelemetryWrapper.java]: it should be ok to make it to be as a *noun*.

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java (hunk 3 above)
Comment (daoshengmu) [reply app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: Done. Thanks.

**Comment by daoshengmu** [comment]
Hope we can get @pocmo 's review before we merge it.

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 4]
```diff
@@ -471,6 +476,18 @@ public void run() {
...
+            return;
+        }
+
+        long elapsedImmersive = SystemClock.elapsedRealtime() - startImmersiveTime;
+        if (elapsedImmersive > MIN_IMMERSIVE_TIME) {
+            Log.i(LOGTAG, "Send immersive time spent to histogram.");
+            TelemetryWrapper.addImmersiveToHistogram(elapsedImmersive);
+        }
+    }
+
```
Comment (pocmo) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: nit: In other projects I'd recommend to move all this into TelemetryWrapper and just expose start/upload methods and not have any logic for this in the activity.

Comment (daoshengmu) [reply app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: Ok. suggestion for hide the logic into TelemetryWrapper. I will fix it.

**Pull request merged**